libopenfpga/libfpgaaesencryption/test/AES_main.cpp

```diff
@@ -0,0 +1,211 @@
+    } //  cout << endl << endl;
+
+
+    // extend key
+    unsigned char expandedKey[ExtdCipherKeyLenghth_max];
+    KeyExpansion(key, Nk, expandedKey);
+	/*  cout << "expandedKey = ";
+    for (unsigned short i = 0; i < ExtdCipherKeyLenghth; i++) { cout << dec << (unsigned short)expandedKey[i] << " "; }
+    cout << "<=> ";
+    for (unsigned short i = 0; i < ExtdCipherKeyLenghth; i++) { cout << "0x" << hex << (unsigned short)expandedKey[i] << " "; }
+    cout << endl << endl;
+	*/
+
+
+	//Read Input Bitstream
+	streampos input_file_size;
+	char *data_in_file;
+	unsigned int plaintext[16];
+	int dec_array_size=0;
+	int dec_file_size;
+	int dec_data;
+	ifstream file ("/research/ece/lnis/USERS/sampathkumar/OpenFPGA/openfpga_flow/tasks/fpga_bitstream
+					/generate_bitstream/device_auto/run001/k6_frac_N10_tileable_40nm/and2/MIN_ROUTE_CHAN_WIDTH/fabric_bitstream.txt",ios::in|ios::ate);
+	if (file.is_open())
+	{
+		input_file_size=file.tellg();
+		data_in_file=new char [input_file_size];
+		file.seekg(0,ios::beg);
+		file.read(data_in_file,input_file_size);
+		file.close();
+	}
+	else cout<<"Unable to open file";
+
+	std::string data_in_string,new_s;
+	std::stringstream ss;
+	ss << data_in_file;
+	data_in_string = ss.str();
+	char Byte_data[4];
+	int bit_count=0;
+	dec_file_size=(input_file_size/8) +1;
+	int dec_array[dec_file_size];
+	int w[dec_file_size];
+	while(int rd_data_cnt<input_file_size)
+	{
+        while(bit_count<4)
+        {
+			if(data_in_string[rd_data_cnt]=='1')
+			{
+				Byte_data[bit_count]=data_in_string[rd_data_cnt];
+				bit_count=bit_count+1;
+				rd_data_cnt++;
+			}
+			if(data_in_string[rd_data_cnt]=='0')
+			{
+
+				Byte_data[bit_count]=data_in_string[rd_data_cnt];
+				bit_count=bit_count+1;
+				rd_data_cnt++;
+			}
+    			else
+			rd_data_cnt++;
+		}
+		new_s = convertToString(Byte_data,4);
+		stringstream ssn(new_s);
+		int binary_data=0;
+		ssn>>binary_data;
+		bit_count=0;
+		dec_data=binaryToDecimal(binary_data);
+		dec_array[dec_array_size]=dec_data;
+		dec_array_size++;
+
+	}
+
+	int arr_size=((sizeof dec_array) / (sizeof *dec_array));	//size of the array
+	int input_data_cnt=0,cipher_array_size=0;;
+	while(input_data_cnt<arr_size)
+	{
+		for (int i=0;i<16;i++)
+		{
+			if(input_data_cnt<arr_size)
+			{
+				plaintext[i]=dec_array[input_data_cnt];
+				input_data_cnt++;
+			}
+			else
+				plaintext[i]=0;
+
+		}
+		
+		// encrypt
+		unsigned char ciphertext[stt_lng];
+		AES_Encrypt(plaintext, expandedKey, Nr, ciphertext);
+		for (unsigned short i=0;i<stt_lng;i++)
+		{
+			w[cipher_array_size]= (unsigned short)ciphertext[i];
+			cipher_array_size++;
+		}
+
+		/*  
+		cout << "ciphertext = ";
+		for (unsigned short i = 0; i <stt_lng; i++) { cout << dec << (unsigned short)ciphertext[i] << " "; }
+		cout << "<=> ";
+		for (unsigned short i = 0; i < stt_lng; i++) { cout << "0x" << hex << (unsigned short)ciphertext[i] << " "; }
+		cout << endl << endl;
+		*/
+		
+	}
+	
+	//Output the Encrypted data in a File
```

Comment:
Similar to the file parser, file writer codes should be placed in separated function, which reads a data base and output the content to a file.

Reply:
I agree, I will do it.